### PR TITLE
Check for plugin update on vagrant startup

### LINKED ISSUE #132: Autodetect new version of the provider and provide info message to update
We should check rubygems, but if this fails, we should just ignore it

### PR DESCRIPTION
**Pull request by ggiamarchi**
Fix #132

### CODE REVIEW

File: source/lib/vagrant-openstack-provider/version_checker.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      attr_accessor :status
+
+      def initialize
+        @status = nil
+      end
+
+      #
+      # Check the latest version from rubygem and set the status
+      #
+      def check
```
Comment (julienvey) [inline source/lib/vagrant-openstack-provider/version_checker.rb]: This method should have a global timeout, of preference very short.

We should also add a global rescue, and just do nothing if anything fails. For instance if rubygems is not available, if we are running the provider on a private corporate network, this check should not degrade the user experience

Comment (ggiamarchi) [inline source/lib/vagrant-openstack-provider/version_checker.rb]: I agree for the timeout. The global rescue is done in the module method who wrap this call.

Comment (ggiamarchi) [inline source/lib/vagrant-openstack-provider/version_checker.rb]: Timeout is now fixed to 3 seconds [here](https://github.com/ggiamarchi/vagrant-openstack-provider/pull/197/files#diff-aebd405ac5da345cd32ad47ba4b50da8R68)

**Comment by ggiamarchi** [comment]
Errors are now fixed.

**Pull request merged**